Account Settings: Fix Overlap with Profile Links on Mobile

torres126 opened the pull request:
#### Changes proposed in this Pull Request

Adds some padding to the bottom of the "Profile Links" box on mobile devices, so that the inline help icon no longer overlaps with buttons. 

#### Testing instructions

Ensure that there's no overlaps and no unintended changes at https://wordpress.com/me, meaning you can click all buttons. 

**Current:**

![beforedgfdfgfgdfgd](https://user-images.githubusercontent.com/43215253/50550091-796c1c00-0c61-11e9-8bdf-0ec5d39c044f.png)

![beforefsdgfdfdggdf](https://user-images.githubusercontent.com/43215253/50550096-8557de00-0c61-11e9-9b75-783fe18a2829.png)

**Intended Proposal:**

![after_sgfdgfdgfs](https://user-images.githubusercontent.com/43215253/50550104-9f91bc00-0c61-11e9-86d4-ecba997e79fd.png)

![after_dfgdfggdf](https://user-images.githubusercontent.com/43215253/50550117-bf28e480-0c61-11e9-8e00-fa25eb7257a9.png)


**After when an error appears:**

![with_an_error_dfgfddgfgdf](https://user-images.githubusercontent.com/43215253/50551126-42533600-0c74-11e9-8722-7c961276c7e7.png)



<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

(cc @ebinnion, @dmsnell & @tyxla) 

Fixes #26257

tyxla commented on client/me/profile-links/index.jsx:
```diff
@@ -168,7 +168,7 @@ class ProfileLinks extends React.Component {
 	render() {
 		return (
-			<div>
+			<div className="me-profile-links">
```
This classname doesn't seem right. As per our [classname recommendations](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#classes), it should have the component name as a prefix (which in our case is `profile-links__`), and then have some sort of suffix - if that's the main wrapper, we often add `main` as a suffix.

tyxla commented on client/me/profile-links/style.scss:
```diff
@@ -1,3 +1,5 @@
+/** @format */
```
I think the Prettier pragmas are no longer necessary after #29403 (also see #29404)

torres126 commented:
Thanks! Updated the class name and removed the Prettier pragma. I tested a few places across the Profile, Reader and My Sites section, but this seems to be the only one where at the furthest point of possible scrolling, the inline help button is overlapping something significant, such as the buttons here. I might be missing somewhere though.

tyxla commented:
Oh well, I'm certain there will be more cases like this. A quick test I did showed some:

**Sharing settings page on Galaxy S5:**
![](https://cldup.com/-XmpfHxA6J.png)

**Comments page on iPhone 5/SE:**
![](https://cldup.com/mWujjdnWfM.png)

Seems like we need a universal solution, which is why I recommended asking for design feedback.

torres126 commented:
Ahh, got it. Just to confirm, who should be pinged for design feedback? (I thought it was a label :sweat_smile: )

tyxla commented:
I'm sure anyone from @Automattic/design would be happy to chime in with feedback on this one 😉

folletto commented:
My suggestion for a _quick fix_ would be to add padding at the end of the page instead at the end of cards. Changing the padding inside a card would modify the proportions of the card thus breaking both the balance of the card itself, as well as the standard padding for the cards.

tyxla commented:
Or perhaps instead of blank space at the bottom we could use the list end component that we display the bottom of the posts page:

![](https://cldup.com/Z0UWZUpFmt.png)

or something similar.

torres126 commented:
Hi @tyxla, I found this PR again on my list of PRs and I think I've caught up with this again (and thankfully, my knowledge of things has increased a lot more since I first opened this PR!). 

Would you mind giving this another spin now?

<img width="317" alt="Screenshot 2019-04-03 at 18 08 44" src="https://user-images.githubusercontent.com/43215253/55499102-1366aa80-563d-11e9-9b14-843ee67e6c80.png">

I quite favour the List End approach, it removes the requirement for additional styling and fixes the issue.

tyxla commented on client/me/profile-links/index.jsx:
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
+							onClosePopoverMenu={ this.closePopoverMenu }
+						/>
+					</SectionHeader>
+					<Card>{ !! this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
```
This `!!` seems to be unnecessary.

tyxla commented on client/me/profile-links/index.jsx:
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
-				<Card>{ this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
-			</div>
+			<Fragment>
+				<div>
```
This `<div />` seems to be unnecessary and can be removed.

tyxla commented on client/me/profile-links/index.jsx:
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
+					<QueryProfileLinks />
+					<SectionHeader label={ this.props.translate( 'Profile Links' ) }>
+						<AddProfileLinksButtons
+							showingForm={ !! this.state.showingForm }
```
Is there any benefit from this `!!`?

torres126 replied:
Not sure how that got there tbh. 🤔 

Oh well, nice spot, addressed in the last commit!

dmsnell commented on client/me/profile-links/index.jsx:
```diff
@@ -173,20 +172,21 @@ class ProfileLinks extends React.Component {
-				<Card>{ this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
-			</div>
+					<SectionHeader label={ this.props.translate( 'Profile Links' ) }>
+						<AddProfileLinksButtons
```
Looks like some extraneous indentation snuck in here. Have you been able to get the git hooks setup @torres126? They should auto-format on save.

torres126 replied:
Hmm, nope, I'm not aware what that is. I'll look into that when I get the chance. :) 

(Before a few days ago, my PRs were made on my tablet which kept erasing lines and I couldn't use tab, so I think many of my PRs may have some odd spacing as a result too)

dmsnell replied:
I'm not sure why they aren't working as they should be Automattic. The hooks are in `.git/hooks`

Can you verify that they are executable?
Are you on Windows?
Are you committing via the `git` command line utility or using some GUI?

torres126 replied:
> Can you verify that they are executable?

Me isn't sure how to do that :sweat_smile: 

> Are you on Windows?

Android for all my previous PRs until a few days ago, where I'm on Mac

> Are you committing via the git command line utility or using some GUI?

Directly on github.com

dmsnell replied:
That makes sense then if you've been editing code through the browser. As you get into a workflow on macOS the auto-formatting should run on `git commit`
